main.go

```diff
@@ -641,10 +650,18 @@ func setupReconcilers(ctx context.Context, mgr ctrl.Manager, watchNamespaces map
 		setupLog.Error(err, "Unable to create controller", "controller", "Machine")
 		os.Exit(1)
 	}
+	machineSetPreflightChecksSet := sets.Set[clusterv1.MachineSetPreflightCheck]{}
+	for _, c := range machineSetPreflightChecks {
+		if c == "" {
+			continue
+		}
+		machineSetPreflightChecksSet.Insert(clusterv1.MachineSetPreflightCheck(c))
```

Comment:
q: should we validate that the users picks only one of the allowed values before adding it to the list?